### PR TITLE
Bootstrap

### PR DESCRIPTION
**Pull request by amine3**

### CODE REVIEW

File: agent/openvas.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+import gvm
+from gvm.protocols import gmp as openvas_gmp
+from gvm import transforms
+
+logger = logging.getLogger(__name__)
+
+ALL_IANA_ASSIGNED_TCP_UDP = '4a4717fe-57d2-11e1-9a26-406186ea4fc5'
+GVMD_FULL_FAST_CONFIG = 'daba56c8-73ec-11df-a475-002264764cea'
+OPENVAS_SCANNER_ID = '08b69003-5fc2-4037-a479-93b440211c73'
```
Comment (deadly-panda) [inline agent/openvas.py:16]: this is always a const ?

File: agent/openvas.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+GMP_USERNAME = 'admin'
+GMP_PASSWORD = 'admin'
+WAIT_TIME = 30
+
+class OpenVas:
+    """OpenVas wrapper to enable using openvas scanner from ostorlab agent class."""
+    def start_scan(self, ip):
```
Comment (deadly-panda) [inline agent/openvas.py:23]: Output type hint, and update docstring.

File: agent/openvas.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            logger.debug('Creating target')
+            target_id = self._create_target(gmp, ip, ALL_IANA_ASSIGNED_TCP_UDP)
+            logger.debug('Creating task for target %s', target_id)
+            task_id = self._create_task(gmp, ip, target_id, GVMD_FULL_FAST_CONFIG, OPENVAS_SCANNER_ID,)
+            logger.debug('Creating report for task %s', task_id)
+            report_id = self._start_task(gmp, task_id)
+            logger.info('Started scan of host %s. Corresponding report ID is %s', str(ip), str(report_id))
+            return task_id
+
+    def _create_target(self, gmp, ip, port_list_id):
```
Comment (deadly-panda) [inline agent/openvas.py:42]: type hints

File: agent/openvas.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            gmp: GMP object.
+            task_id: task id.
+
+        Returns:
+            - task result.
+        """
+        response = gmp.start_task(task_id)
+        return response[0].text
+
+    def wait_task(self, task_id):
```
Comment (deadly-panda) [inline agent/openvas.py:87]: type

File: agent/openvas.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                        logger.debug('Checking task %s', task.xpath('@id')[0])
+                        if task.xpath('@id')[0] == task_id:
+                            logger.info('Scan progress %s', str(task.find('progress').text))
+                            if task.find('status').text == 'Done':
+                                return True
+                except socket.timeout:
+                    logger.error('Socket timeout error')
+                time.sleep(WAIT_TIME)
+
+    def get_results(self):
```
Comment (deadly-panda) [inline agent/openvas.py:106]: same

File: agent/openvas_agent.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+from agent import openvas
+
+logging.basicConfig(
+    format='%(message)s',
+    datefmt='[%X]',
+    handlers=[rich_logging.RichHandler(rich_tracebacks=True)]
+)
+logger = logging.getLogger(__name__)
+logger.setLevel('DEBUG')
```
Comment (deadly-panda) [inline agent/openvas_agent.py:21]:
```suggestion
```

File: ostorlab.yaml [hunk 7]
```diff
@@ -1,9 +1,10 @@
...
-in_selectors: # List of input selectors.
-  - v3.healthcheck.ping
-out_selectors: [] # List of output selectors.
-docker_file_path : Dockerfile # Dockerfile path for automated release build.
-docker_build_root : . # Docker build dir for automated release build.
+name: openvas
+version: 0.0.1
+description: "OpenVas is an open-source security scanner part of the Greenbone Vulnerability Security Suite."
+in_selectors:
+  - v3.asset.ip
```
Comment (deadly-panda) [inline ostorlab.yaml:6]:
```suggestion
  - v3.asset.ip.v4
  - v3.asset.ip.v6
```

File: requirement.txt [hunk 8]
```diff
@@ -1,2 +1,4 @@
 ostorlab
-rich
\ No newline at end of file
+rich
```
Comment (deadly-panda) [inline requirement.txt:2]:
```suggestion
```

Comment (deadly-panda) [reply requirement.txt:2]: Isn't this part of ostorlab ?

Comment (3asm) [reply requirement.txt:2]: But explicity used separately. If we move out, we need to keep it.

File: ostorlab.yaml [hunk 9]
```diff
@@ -1,9 +1,10 @@
...
-version: 0.0.0
-description: Agent description. # Support for Markdown format.
-in_selectors: # List of input selectors.
-  - v3.healthcheck.ping
-out_selectors: [] # List of output selectors.
-docker_file_path : Dockerfile # Dockerfile path for automated release build.
-docker_build_root : . # Docker build dir for automated release build.
+name: openvas
+version: 0.0.1
+description: "OpenVas is an open-source security scanner part of the Greenbone Vulnerability Security Suite."
```
Comment (3asm) [inline ostorlab.yaml:4]: Please align the markdown.

File: scripts/sync.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
+from gvm.connections import TLSConnection
+from gvm.protocols.gmp import Gmp
+from gvm.transforms import EtreeTransform
+from gvm.xml import pretty_print
+
+connection = TLSConnection(hostname='127.0.0.1', port=9390)
```
Comment (deadly-panda) [inline scripts/sync.py:6]:
```suggestion
connection = TLSConnection(hostname='localhost', port=9390)
```

Comment (deadly-panda) [reply scripts/sync.py:6]: Since u r using localhost in other parts.

File: agent/openvas_agent.py [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+START_SCRIPT = '/start.sh'
+LOG_FILE = '/usr/local/var/log/gvm/gvmd.log'
+VT_CHECK = b'Updating VTs in database ... done'
+WAIT_VT_LOAD = 30
+
+class OpenVasAgent(agent.Agent, agent_report_vulnerability_mixin.AgentReportVulnMixin):
+    """OpenVas Agent."""
+
+    def start(self) -> None:
+        """Calls that start.sh script to bootstrap the scanner."""
```
Comment (3asm) [inline agent/openvas_agent.py:32]:
```suggestion
        """Calls the start.sh script to bootstrap the scanner."""
```

File: tests/conftest.py [hunk 12]
```diff
@@ -1,10 +1,17 @@
...
-"""
+from ostorlab.agent import message
 
-# import pytest
+
+@pytest.fixture
+def scan_message():
+    """Creates a dummy message of type v3.asset.ip to be used by the agent for testing purposes.
+    """
+    selector = 'v3.asset.ip'
```
Comment (deadly-panda) [inline tests/conftest.py:11]:
```suggestion
    selector = 'v3.asset.ip'
```
```suggestion
    selector = 'v3.asset.ip.v4'
```

File: tests/openvas_agent_test.py [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+            utils_definitions.Arg(name='reporting_engine_base_url', type='str', value=b'https://toto.ostorlab.co/test'),
+            utils_definitions.Arg(name='reporting_engine_token', type='str', value=b'123456')],
+        healthcheck_port=5301)
+    mocker.patch('agent.openvas.OpenVas.start_scan',return_value='hduzehfuhehfuhef')
+    mocker.patch('agent.openvas.OpenVas.wait_task',return_value=None)
+    mocker.patch('agent.openvas.OpenVas.get_results',return_value='xml_result')
+    mock_report_vulnerability = mocker.patch('agent.openvas_agent.OpenVasAgent.report_vulnerability', return_value=None)
+    test_agent = openvas_agent.OpenVasAgent(definition, settings)
+    test_agent.process(scan_message)
+    mock_report_vulnerability.assert_called()
```
Comment (deadly-panda) [inline tests/openvas_agent_test.py:27]:
```suggestion
    test_agent = openvas_agent.OpenVasAgent(definition, settings)
    
    test_agent.process(scan_message)
    
    mock_report_vulnerability.assert_called()
```

File: agent/openvas_agent.py [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+        logger.info('processing message')
+        openvas_wrapper = openvas.OpenVas()
+        task_id = openvas_wrapper.start_scan(message.data.get('host'))
+        openvas_wrapper.wait_task(task_id)
+        result = openvas_wrapper.get_results()
+        self._send_results(result)
+        logger.info('Scan finished.')
+
+    def _wait_vt_ready(self):
+        while True:
```
Comment (3asm) [inline agent/openvas_agent.py:47]: docstring and explain why we are doing it.

File: agent/openvas_agent.py [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+            with open(LOG_FILE, 'rb') as f:
+                for line in f.readlines():
+                    if VT_CHECK in line:
+                        return True
+            logger.info('Waiting for VT to load in database.')
+            time.sleep(WAIT_VT_LOAD)
+
+    def _send_results(self, results):
+        self.report_vulnerability(
```
Comment (3asm) [inline agent/openvas_agent.py:56]: Docstring and needs to be updated.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: tests/openvas_agent_test.py [hunk 16]
```diff
@@ -0,0 +1,51 @@
+"""Unittests for nuclei class."""
```
Comment (3asm) [inline tests/openvas_agent_test.py:1]:
```suggestion
"""Unittests for openvas agent class."""
```

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**